src/document_add.c

```diff
@@ -161,6 +161,10 @@ static int doAddDocument(RedisModuleCtx *ctx, RedisModuleString **argv, int argc
     RedisModule_ReplyWithError(ctx, "Unknown index name");
     goto cleanup;
   }
+  if (canBlock) {
```

Comment:
Move lines 155 here, and then:

`canBlock = !sp->temporary && CheckConcurrentSupprt(...)`